ws-server/src/server.rs

```diff
@@ -21,477 +21,296 @@
 // SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
 // CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 // OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR
-// IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// IN background_task WITH THE SOFTWARE OR THE USE OR OTHER
 // DEALINGS IN THE SOFTWARE.
 
-use crate::raw::{RawServer, RawServerEvent, RawServerRequestId, RawServerSubscriptionId};
-use crate::transport::WsTransportServer;
-use jsonrpsee_types::{
-	error::Error,
-	jsonrpc::{self, JsonValue},
-};
-
-use futures::{channel::mpsc, future::Either, pin_mut, prelude::*};
+use futures::io::{BufReader, BufWriter};
+use jsonrpsee_types::{error::Error, jsonrpc::SubscriptionId};
 use parking_lot::Mutex;
-use std::{
-	collections::{HashMap, HashSet},
-	convert::TryFrom,
-	error,
-	net::SocketAddr,
-	sync::{atomic, Arc},
+use rustc_hash::FxHashMap;
+use serde::{Deserialize, Serialize};
+use serde_json::value::{to_raw_value, RawValue};
+use soketto::handshake::{server::Response, Server as SokettoServer};
+use std::net::SocketAddr;
+use std::sync::Arc;
+use thiserror::Error;
+use tokio::{
+	net::{TcpListener, ToSocketAddrs},
+	sync::mpsc,
 };
+use tokio_stream::{wrappers::TcpListenerStream, StreamExt};
+use tokio_util::compat::TokioAsyncReadCompatExt;
+
+use crate::types::{JsonRpcError, JsonRpcErrorParams};
+use crate::types::{JsonRpcInvalidRequest, JsonRpcRequest, JsonRpcResponse, TwoPointZero};
+use crate::types::{JsonRpcNotification, JsonRpcNotificationParams};
+
+type ConnectionId = usize;
+
+type RpcSender<'a> = &'a mpsc::UnboundedSender<String>;
+type RpcId<'a> = Option<&'a RawValue>;
+type Method = Box<dyn Send + Sync + Fn(RpcId, RpcParams, RpcSender, ConnectionId) -> anyhow::Result<()>>;
```

Comment:
as I said before `Box<dyn Trait>` is syntactic sugar for `Box<dyn Trait + 'static>`, the proper way to deduce the `lifetime` is to assign a lifetime to it such as `type Method<'a> = Box<dyn Send + Sync + 'a>`,

Then I think we could get rid of the `'static` bound on the closures but fine for now.